charts/apisix/templates/_pod.tpl

```diff
@@ -62,7 +62,11 @@ spec:
         {{- if (gt (len .tcp) 0) }}
         {{- range $index, $port := .tcp }}
         - name: proxy-tcp-{{ $index | toString }}
+        {{- if kindIs "map" $port }}
+          containerPort: {{ $port.addr }}
```

Comment:
`addr` might contains IP, so here we cannot treat it as a port, we may need to use `splitList` to fetch the port from the `addr`.

Reply:
@cr7258 PTAL